### PR TITLE
Added deployment of Test contract for Nexus

### PR DESCRIPTION
**Pull request by sasa-ethernal**
# Description

Please provide a detailed description of what was done in this PR

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

Please complete this section if you ran manual tests for this functionality, otherwise delete it

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: consensus/polybft/contractsapi/init.go [hunk 1]
```diff
@@ -354,6 +354,12 @@ func init() {
 		log.Fatal(err)
 	}
 
+	// Nexus contracts
+	ClaimsTest, ERC1967Proxy, err = initNexusContracts()
```
Comment (igorcrevar) [inline consensus/polybft/contractsapi/init.go]: Move this initialization somewhere in e2e test framework. its only part of the test

File: consensus/polybft/contractsapi/init_nexus.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package contractsapi
```
Comment (igorcrevar) [inline consensus/polybft/contractsapi/init_nexus.go]: Move this initialization file  somewhere in e2e test framework. its only part of the test

File: consensus/polybft/contractsapi/nexus-artifacts-gen/main.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+const (
+	extension = ".sol"
+)
+
+func main() {
+	_, filename, _, _ := runtime.Caller(0) //nolint: dogsled
+	currentPath := path.Dir(filename)
+	//nolint: godox
+	// TODO:Nexus update path
```
Comment (igorcrevar) [inline consensus/polybft/contractsapi/nexus-artifacts-gen/main.go:26]: Can we remove this TODO if not needed?

**Review by Dejan-Nedic — APPROVED** [review]

File: e2e-polybft/cardanofw/test-apex-bridge.go [hunk 4]
```diff
@@ -306,6 +306,10 @@ func RunApexBridge(
 
 	fmt.Printf("Apex bridge setup done\n")
 
+	SetupAndRunNexusBridge(t, bladeValidatorsNum)
```
Comment (igorcrevar) [inline e2e-polybft/cardanofw/test-apex-bridge.go]: Nexus is not a bridge. It is just a chain, similar like prime and vector. The only difference is that it is evm and not cardano. Rename all the structs and funcs accordingly. Nexus should be started before apex bridge (can be after  prime and vector)

**Review by Dejan-Nedic — APPROVED** [review]

**Review by Dejan-Nedic — DISMISSED** [review]

**Pull request merged**